src/lib/builders/color-picker/create.ts

```diff
@@ -0,0 +1,960 @@
+
+            return {
+                'aria-label': 'Button on color canvas, used to select the saturation and brightness.',
+                style: styleToString({
+                    position: 'absolute',
+                    top: `${top}px`,
+                    left: `${left}px`,
+                    'background-color': `${$value}`
+                })
+            }
+        },
+        action: (node: HTMLButtonElement) => {
+            const rect = node.getBoundingClientRect();
+
+            colorPickerDims.set({
+                width: rect.width,
+                height: rect.height
+            });
+
+            const unsubEvents = executeCallbacks(
+                addMeltEventListener(node, 'mousedown', () => {
+                    dragging = true;
+                }),
+                addMeltEventListener(node, 'mouseup', () => {
+                    dragging = false;
+                }),
+                addMeltEventListener(node, 'keyup', (e) => {
+                    const { key } = e;
+                    const keys = ['ArrowUp', 'ArrowDown', 'ArrowLeft', 'ArrowRight'];
+
+                    if (!keys.includes(key)) return;
+
+                    keyDurations[<ArrowKeys>key] = null;
+                }),
+                addMeltEventListener(node, 'keydown', (e) => {
+                    const keys = ['ArrowUp', 'ArrowDown', 'ArrowLeft', 'ArrowRight'];
+
+                    if (!keys.includes(e.key)) return;
+
+                    const key = <ArrowKeys>e.key;
+
+                    e.preventDefault();
+
+                    let duration = 0;
+
+                    if (!keyDurations[key]) {
+                        keyDurations[key] = Date.now();
+                    } else {
+                        duration = Date.now() - <number>keyDurations[key];
+                    }
+
+                    const step = duration > 1000 ? speedUpStep : 1;
+
+                    const { x, y } = get(colorPickerPos);
+                    const { height: canvasH, width: canvasW } = get(colorCanvasDims);
+
+                    // Move the picker button and restrict movement to within the canvas.
+                    if (key === 'ArrowUp' && y - 1 >= 0) {
+                        colorPickerPos.set({ x, y: Math.max(y - step, 0) });
+                    } else if (key === 'ArrowDown' && y + 1 <= canvasH) {
+                        colorPickerPos.set({ x, y: Math.min(y + step, canvasH) });
+                    } else if (key === 'ArrowRight' && x + 1 <= canvasW) {
+                        colorPickerPos.set({ x: Math.min(x + step, canvasW), y });
+                    } else if (key === 'ArrowLeft' && x - 1 >= 0) {
+                        colorPickerPos.set({ x: Math.max(x - step, 0), y });
+                    }
+                })
+            );
+
+            return {
+                destroy() {
+                    unsubEvents();
+                },
+            }
+        }
+    });
+
+    const hueSlider = builder(name('hue-slider'), {
```

Comment:
Would be cool to have keyboard events on the sliders as well 😄

Reply:
@TGlide ah, you mean that if the slider is focused, rather than just the slider button?